esphome/components/wake_on_lan/button.py

```diff
@@ -0,0 +1,32 @@
+import esphome.codegen as cg
+from esphome.components import button
+import esphome.config_validation as cv
+from esphome.const import CONF_ID
+
+CONF_TARGET_MAC_ADDRESS = "target_mac_address"
+
+wake_on_lan_ns = cg.esphome_ns.namespace("wake_on_lan")
+
+WakeOnLanButton = wake_on_lan_ns.class_("WakeOnLanButton", button.Button, cg.Component)
+
+DEPENDENCIES = ["wifi"]
```

Comment:
In case someone uses ethernet instead of wifi
```suggestion
DEPENDENCIES = ["network"]
```